### ISSUE TITLE
Message for the `sys.missing_fields` error is not translated

### ISSUE THREAD

**Issue by abbyad** [medic/medic-webapp#3923]
When accepting a patient report the `sys.missing_fields` error message is added to the doc if the report is missing a required field. Although the error is added to doc, the `errors[].message` field is not translated.

Here is [an example](https://standard.dev.medicmobile.org/medic/_design/medic/_rewrite/#/reports/82c1c868215daa278f45209ab73ca2ac):
```json
[
   {
       "code": "sys.missing_fields",
       "fields": [
           "delivery_code"
       ],
       "message": "sys.missing_fields"
   },
   {
       "code": "invalid_delivery_code",
       "message": "The delivery code was missing. Please use code 'F' for facility, 'S' for homebirth with Skilled Birth Attendant, 'NS' for birth with no skilled attendant."
   }
]
```
and the translation:
```
sys.missing_fields = Missing or invalid fields\: {{fields}}.
```

It is correct that the only outgoing message is the `invalid_delivery_code` message, but showing the missing fields error is confusing because it is not translated, as seen here: 
![image](https://user-images.githubusercontent.com/789512/30562172-8adc87e6-9c8b-11e7-8790-89f4928940b1.png)

We should make sure that all error messages are translated.

**Comment by sglangevin** [comment]
Is this a new bug? If not, then I'm hesitant to hold up the release. It can be fixed in the next release.

**Comment by abbyad** [comment]
Just tested on 2.12.5 and was able to reproduce... so it is not a new bug. Makes sense to fix this later.

**Comment by sglangevin** [comment]
Cool, I am going to move this one to 2.14.0.

**Comment by garethbowen** [comment]
This is due to the translation function used in couchdb looking for the translation in the ddoc app_settings which has been deprecated in favour of using translation documents (#1706). It's difficult to fix because couchdb functions don't have unrestricted access to the database so can't just fetch the translation docs like other parts of the code.

One workaround is to modify the ddoc.app_settings to have the translations, eg:

```json
{
  "app_settings": {
   "translations": [
       {
           "key": "sys.missing_fields",
           "translations": [
               {
                   "locale": "en",
                   "content": "Missing or invalid fields. {{fields}}"
               }
           ]
       }
   ]
  }
}
```

This will be fixed when we move away from using kanso packages (#3019 - currently slated for 3.0.0) at which point the `/add` functionality will be implemented in API which has access to the regular translation function.

@abbyad Is the proposed workaround sufficient until then?

**Comment by abbyad** [comment]
Ok, good to know. We'll manually add that until it is resolved.

Also, this should be considered as part of https://github.com/medic/medic-webapp/issues/3923, and this is certainly a duplicate of https://github.com/medic/medic-webapp/issues/3368, which I'll be closing since this one has better info.

**Comment by garethbowen** [comment]
It can't be done as part of #3190 for the same reason - this code can't access the translations without a bunch of workaround effort.

**Comment by abbyad** [comment]
I had presumed #3190 wasn't happening before #3019 anyway, so was grouping those two logically as parts of app_settings that have to have the translations directly in app_settings.

**Comment by garethbowen** [comment]
As discussed, moving to 3.0.0.

**Comment by garethbowen** [comment]
Blocked on #3019 - is fixed for free once that's done.

**Comment by ngaruko** [comment]
Tested on `beta.dev`. 
Sent a delivery report with `*` as code. 
![image](https://user-images.githubusercontent.com/6979995/38913376-862ac362-432e-11e8-8993-1bbf3c1de360.png)

Report submitted with proper errors/translations...
![image](https://user-images.githubusercontent.com/6979995/38913401-a74c32ce-432e-11e8-910a-dea2dd2a4fbc.png)

 **LGTM**
